lib/board.rb

```diff
@@ -18,18 +18,30 @@ def full?
         positions.all? {|position| position != Token::EMPTY }
     end
 
-    def position_available?(position)
-        positions[position] == Token::EMPTY
-    end
-
     def empty_positions
-        positions.each_with_index.reduce([]) do |empty_positions, (_position, idx)| 
-            empty_positions.push(idx) if position_empty?(idx)
+        positions.each_with_index.reduce([]) do |empty_positions, (_position, index)| 
```

Comment:
I had thought a `map` would iterate over each item and transform it in some way. In this case, I'm looking through and picking out matching items so it felt like `each` or even `find` would be more appropriate?

Reply:
Wouldn't select be better, then?

Reply:
I thought of using select, but I need to know each index in the array, and after filtering the values, only return the matching index. Which is why i did the `each_with_index` and then `reduce` to specify what I wanted to return

Reply:
Lol look at me suggesting map because I am cutting back on coffee.
But `positions.each_index.select` could be pretty useful!

Reply:
Just to make sure I'm doing this right -- when I use select instead of reduce, i can't control what i'm returning. AKA, it returns the entire unit of position/index instead of just the index.
Was there another way to use select to return the desired value?

So this function
```
   # positions is an array
    def empty_positions
        positions.each_with_index.select() do |_position, index| 
            index if position_empty?(index)
        end
    end
```

returns 
<img width="528" alt="Screen Shot 2019-05-06 at 7 13 13 AM" src="https://user-images.githubusercontent.com/29721784/57230979-7053d880-6fce-11e9-884f-d9c6bf67d78e.png">

Reply:
Ah!  Okay, select alone probably doesn't work.  If you're filtering _and_ mapping then `reduce` is totally your huckleberry.